fvm/environment/programs.go

```diff
@@ -311,17 +327,16 @@ func (s *dependencyStack) push(loc common.Location) {
 func (s *dependencyStack) add(dependencies derived.ProgramDependencies) {
 	l := len(s.trackers)
 	if l == 0 {
-		// stack is empty.
-		// This is expected if loading a program that is already cached.
-		return
+		// This cannot happen, as the root of the stack is always present.
+		panic("Dependency stack unexpectedly empty")
 	}
 
 	s.trackers[l-1].dependencies.Merge(dependencies)
 }
 
 // pop the last dependencies on the stack and return them.
 func (s *dependencyStack) pop() (common.Location, derived.ProgramDependencies, error) {
-	if len(s.trackers) == 0 {
+	if len(s.trackers) <= 1 {
```

Comment:
can you explain this line of change? why `==1` is invalid?

It would be confusing when `len(s.trackers) == 1` while the error message says "because it is empty", we should at least log the `len(s.trackers)`.

Reply:
there should always be at least one entry in the stack. that entry keeps a dependencies of the script/transaction that is being executed.

I agree the error could be better and I will add comments in a new PR.